task/orchestrator.go

```diff
@@ -190,3 +209,15 @@ func (o *Orchestrator) HealthChecks() (_ string, ready, live func(ctx context.Co
 			return nil
 		}, nil
 }
+
+type retryableError struct {
+	error
+}
+
+func newRetryableError(err error) retryableError {
+	return retryableError{err}
+}
+
+func retryableErrorf(format string, a ...any) retryableError {
+	return retryableError{fmt.Errorf(format, a...)}
+}
```

Comment:
Setup for my next PR to add a runner API client for automatic task retries and infrastructure failure events (that make sense for GOAT to handle). That PR should (hopefully) be the last piece for the first cut of GOAT, excluding the integration work with the runner agent.